test_server.py

```diff
@@ -164,77 +164,89 @@ def test_create_recommendation(self):
         self.assertEqual( resp.status_code, HTTP_400_BAD_REQUEST )
 
-    def test_create_pet_with_no_priority_and_related_product_id(self):
+    def test_create_recommendation_with_no_priority_and_related_product_id(self):
         new_recommendation = {'parent_product_id': '2', 'type': 'x-sell'}
         data = json.dumps(new_recommendation)
         resp = self.app.post('/recommendations', data=data, content_type='application/json')
         self.assertEqual( resp.status_code, HTTP_400_BAD_REQUEST )
 
-    def test_create_pet_with_no_priority_and_type(self):
+    def test_create_recommendation_with_no_priority_and_type(self):
         new_recommendation = {'parent_product_id': '2', 'related_product_id':'2'}
         data = json.dumps(new_recommendation)
         resp = self.app.post('/recommendations', data=data, content_type='application/json')
         self.assertEqual( resp.status_code, HTTP_400_BAD_REQUEST )
 
-    def test_create_pet_with_no_related_product_id_and_type(self):
+    def test_create_recommendation_with_no_related_product_id_and_type(self):
         new_recommendation = {'parent_product_id': '2', 'priority': '5'}
         data = json.dumps(new_recommendation)
         resp = self.app.post('/recommendations', data=data, content_type='application/json')
         self.assertEqual( resp.status_code, HTTP_400_BAD_REQUEST )
 
     def test_update_recommendation(self):
-        new_recommendation = {'parent_product_id': '2', 'priority': '5', 'related_product_id':'4', 'type': 'x-sell'}
+        new_recommendation = {'parent_product_id': '2', 'priority': '5', 'related_product_id':'4', 'type': 'up-sell'}
         data = json.dumps(new_recommendation)
         resp = self.app.put('/recommendations/3', data=data, content_type='application/json')
         self.assertEqual( resp.status_code, HTTP_200_OK )
         new_json = json.loads(resp.data)
-        self.assertEqual (new_json['parent_product_id'], 2)
+        self.assertEqual (new_json['type'], 'up-sell')
 
     def test_update_recommendation_with_no_data(self):
         resp = self.app.put('/recommendations/3', data=None, content_type='application/json')
         self.assertEqual( resp.status_code, HTTP_400_BAD_REQUEST )
 
+    def test_update_recommendation_with_no_parent_product_id(self):
+        new_recommendation = {'priority': '5', 'related_product_id':'4', 'type': 'x-sell'}
+        data = json.dumps(new_recommendation)
+        resp = self.app.put('/recommendations/2', data=data, content_type='application/json')
+        self.assertEqual( resp.status_code, HTTP_400_BAD_REQUEST )
+
+    def test_update_recommendation_not_found(self):
+        new_recommendation = {'parent_product_id': '2', 'priority': '5', 'related_product_id':'4', 'type': 'up-sell'}
+        data = json.dumps(new_recommendation)
+        resp = self.app.put('/recommendations/0', data=data, content_type='application/json')
+        self.assertEquals( resp.status_code, status.HTTP_404_NOT_FOUND )
+
```

Comment:
@rohithreddy1  sure
when we discussed the code for PUT it was decided that only type and priority is changable, while  parent_product_id, related_product_id, id will be used as validation. so since related_product_id is wrong (4 to 3), the PUT did not change anything, and returns 200 as because of idempotency.
```
UPDATE recommendations \
                      SET type=\"%s\", priority=%s \
                      WHERE parent_product_id=%s \
                      AND related_product_id=%s AND id=%s"
```